addon/webextension/background/main.js

```diff
@@ -85,22 +151,75 @@ window.main = (function () {
     if (shouldOpenMyShots(url)) {
       return true;
     }
-    if (url.startsWith(backend) || /^(?:about|data|moz-extension):/i.test(url)) {
+    if (isShotOrMyShotPage(url) || /^(?:about|data|moz-extension):/i.test(url) || isBlacklistedUrl(url)) {
       return false;
     }
     return true;
   }
 
+  function isShotOrMyShotPage(url) {
+    // It's okay to take a shot of any pages except shot pages and My Shots
+    if (!url.startsWith(backend)) {
+      return false;
+    }
+    let path = url.substr(backend.length).replace(/^\/*/, "").replace(/#.*/, "").replace(/\?.*/, "");
```

Comment:
It would be nice to use a `URL` object for this... But failing that, the second two regexps can be consolidated into `.replace(/[#?].*/, "")`